app/src/sharedTest/java/org/oppia/app/topic/TopicFragmentTest.kt

```diff
@@ -0,0 +1,342 @@
+package org.oppia.app.topic
+
+import android.app.Application
+import android.content.Context
+import androidx.test.core.app.ActivityScenario.launch
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.action.ViewActions.swipeLeft
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDescendantOfA
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.isRoot
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.CoroutineDispatcher
+import org.hamcrest.CoreMatchers.allOf
+import org.hamcrest.CoreMatchers.containsString
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.recyclerview.RecyclerViewMatcher.Companion.atPositionOnView
+import org.oppia.app.utility.EspressoTestsMatchers.matchCurrentTabTitle
+import org.oppia.app.utility.OrientationChangeAction.Companion.orientationLandscape
+import org.oppia.util.threading.BackgroundDispatcher
+import org.oppia.util.threading.BlockingDispatcher
+import javax.inject.Singleton
+
+/** Tests for [TopicActivity]. */
+@RunWith(AndroidJUnit4::class)
+class TopicFragmentTest {
+  private val topicName = "First Test Topic"
```

Comment:
Let's make this a file-level constant instead of a property, e.g.:

```kotlin
private const val FIRST_TOPIC_NAME = "First Test Topic"

@RunWith(AndroidJUnit4::class)
class TopicFragmentTest {
  ...
}
```

Reply:
Resolved

Reply:
Resolved